src/lib/client.js

```diff
@@ -90,23 +90,47 @@ StreamClient.prototype = {
       });
 
       this.requestAgent = this.baseUrl.startsWith('https://') ? httpsAgent : httpAgent;
-
-      // setup personalization and collections
-      this.personalizationToken = signing.JWTScopeToken(
-        this.apiSecret, 'personalization', '*', {userId: '*', feedId: '*', expireTokens: this.expireTokens });
-      this.collectionsToken = signing.JWTScopeToken(
-        this.apiSecret, 'collections', '*', {userId: '*', feedId: '*', expireTokens: this.expireTokens });
-
-      this.personalization = new Personalization(this);
-      this.collections = new Collections(this);
     }
 
+    this.personalization = new Personalization(this);
+    this.collections = new Collections(this);
+
     /* istanbul ignore next */
     if (this.browser && this.apiSecret) {
       throw new errors.FeedError('You are publicly sharing your App Secret. Do not expose the App Secret in browsers, "native" mobile apps, or other non-trusted environments.');
     }
   },
 
+  getPersonalizationToken: function() {
+    if (this._personalizationToken) {
+      return this._personalizationToken;
+    }
+
+    if (this.apiSecret) {
+        this.personalizationToken = signing.JWTScopeToken(
```

Comment:
You are checking for presence of `_personalizationToken` but here you are assigning to `personalizationToken` (no underscore prefix)
This will lead to generating new token every time the function is called

Reply:
Good catch. Resolved with a squashed commit for clean git history.